src/core/RemotePlugin.cpp

```diff
@@ -54,7 +54,7 @@ ProcessWatcher::ProcessWatcher( RemotePlugin * _p ) :
 
 void ProcessWatcher::run()
 {
-	while( !m_quit && m_plugin->isRunning() )
+	while( !m_quit && (m_plugin->messagesLeft() || m_plugin->isRunning()) )
```

Comment:
It seems to me there's a possibility for a race condition:
1. `m_plugin->messagesLeft()` returns `false`.
2. VST posts new message and then exits
3. `m_plugin->isRunning()` returns `false`.
4. Loop aborts even though there are pending messages.

I think the conditional should just be rearranged so that `m_plugin->messagesLeft()` is only checked _after_ `m_plugin->isRunning()`:
```
while( !m_quit && (m_plugin->isRunning() || m_plugin->messagesLeft()) )
```

I'm not very familiar with the area of the code though, so don't hesitate to point out why I might be wrong about this.

Reply:
heh :) good catch, thanks. (I wanted to write something about why you were wrong but then realized you were right)